catalog/plugin/catalog-plugin-query-monitor-plugin/src/main/java/ddf/catalog/querymonitor/ActiveSearch.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.querymonitor;
+
+import java.util.Date;
+import java.util.UUID;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.source.Source;
+
+/**
+ * {@link ActiveSearch} provides relevant data pertaining to a search that is currently active (a search
+ * that has not yet completed). Includes the client information {@link String}, CQL query {@link String},
+ * start time of the search {@link java.util.Date}, source being searched {@link ddf.catalog.source.Source},
+ * and uniqueID {@link java.util.UUID}
+ */
+
+public class ActiveSearch {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ActiveSearch.class);
+
+    private String cqlQuery = "";
+
+    private Date startTime;
+
+    private Source source;
+
+    private UUID uniqueID = null;
+
+    private String clientInfo = "";
+
+    public String getCQL() {
+        return cqlQuery;
+    }
+
+    /**
+     * Returns a {@link Date} representing the approximate time this search started.
+     *
+     * @return {@link Date} representing the approximate time the search started. Can return {@link null}
```

Comment:
For all the methods that can return null, it would be helpful to document the conditions under which that would happen.